packages/nextjs/components/NetworksDropdown.tsx

```diff
@@ -80,36 +131,115 @@ export const NetworksDropdown = ({ onChange }: { onChange: (options: any) => any
     }
   }, []);
 
+  const handleSelectChange = (newValue: SingleValue<Options> | MultiValue<Options>) => {
+    const selected = newValue as SingleValue<Options>;
+    if (selected?.value === "other-chains") {
+      if (seeAllModalRef.current) {
+        seeAllModalRef.current.showModal();
+      }
+      if (searchInputRef.current) {
+        searchInputRef.current.focus();
+      }
+    } else {
+      setSelectedOption(selected);
+      onChange(selected);
+    }
+  };
+
+  const handleChainSelect = (option: Options) => {
+    const groupName = option.isTestnet ? "testnet" : "mainnet";
+    if (!groupedOptions[groupName].options.some(chain => chain.value === option.value)) {
+      groupedOptions[groupName].options.push(option);
```

Comment:
We have to be a bit careful while mutating the objects in react directly and  not using `setState` function. Probably we should have created an state for `groupedOptions` and then changed with set function. 

But I tested everything and works great! Incase in future we notice some bug related to dropdown not showing correct chains, we could revisit this logic

Reply:
Resolving this for now but will keep this in mind!